modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -3265,9 +3282,31 @@ def broadcast_apply(
                 axis
             ), self.copy_axis_cache(axis)
 
-        new_frame = self._partition_mgr_cls.broadcast_apply(
-            axis, func, left_parts, right_parts
-        )
+        # check the conditions for use of dynamic partitioning
+        use_dynamic_partitioning = False
+        if self.num_parts <= 1.5 * CpuCount.get():
+            use_dynamic_partitioning = True
+
+            # When the frame is large, dynamic partitioning
+            # performs worse than the based approach
+            frame_size = self.size
+            if frame_size and (frame_size >= 4 * 10**9 or len(self) >= 10**7):
```

Comment:
You introduced a new method .size, which avoid triggering computation for row and columns, but then use len(self). My questions are:

1) Should we keep the condition for len(self)?
2) Should we add the condition for len(self.columns)?
3) Should we have the new method .size()

Reply:
I see, thanks! Should we have a check for len(self.columns)?

Reply:
Probably, but I haven't checked because it's a rare case

Reply:
Outdated